diff -Naur gawk-5.2.1/support/cdefs.h gawk-5.2.1.tpg/support/cdefs.h
--- gawk-5.2.1/support/cdefs.h	2022-11-17 16:18:30.000000000 +0000
+++ gawk-5.2.1.tpg/support/cdefs.h	2023-05-06 12:05:08.668027825 +0000
@@ -685,7 +685,7 @@
 #  define __attr_access_none(argno)
 #endif
 
-#if __GNUC_PREREQ (11, 0)
+#if __GNUC_PREREQ (11, 0) && !defined(__clang__)
 /* Designates dealloc as a function to call to deallocate objects
    allocated by the declared function.  */
 # define __attr_dealloc(dealloc, argno) \
