diff -Naur gawk-5.1.1/m4/lib-link.m4 gawk-5.1.1.tpg/m4/lib-link.m4
--- gawk-5.1.1/m4/lib-link.m4	2020-08-28 11:48:05.000000000 +0000
+++ gawk-5.1.1.tpg/m4/lib-link.m4	2021-11-04 09:05:58.409964448 +0000
@@ -382,7 +382,7 @@
           fi
           if test "X$found_dir" != "X"; then
             dnl Found the library.
-            LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-L$found_dir -l$name"
+            LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ } -l$name"
             if test "X$found_so" != "X"; then
               dnl Linking with a shared library. We attempt to hardcode its
               dnl directory into the executable's runpath, unless it's the
@@ -443,7 +443,7 @@
                       fi
                     done
                     if test -z "$haveit"; then
-                      LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ }-L$found_dir"
+                      LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ }"
                     fi
                     if test "$acl_hardcode_minus_L" != no; then
                       dnl FIXME: Not sure whether we should use
@@ -470,7 +470,7 @@
               else
                 dnl We shouldn't come here, but anyway it's good to have a
                 dnl fallback.
-                LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ }-L$found_dir -l$name"
+                LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ } -l$name"
               fi
             fi
             dnl Assume the include files are nearby.
@@ -580,7 +580,7 @@
                         if test -z "$haveit"; then
                           if test -d "$dependency_libdir"; then
                             dnl Really add $dependency_libdir to $LIBNAME.
-                            LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ }-L$dependency_libdir"
+                            LIB[]NAME="${LIB[]NAME}${LIB[]NAME:+ }"
                           fi
                         fi
                         haveit=
@@ -594,7 +594,7 @@
                         if test -z "$haveit"; then
                           if test -d "$dependency_libdir"; then
                             dnl Really add $dependency_libdir to $LTLIBNAME.
-                            LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-L$dependency_libdir"
+                            LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }"
                           fi
                         fi
                       fi
diff -Naur gawk-5.1.1/m4/mpfr.m4 gawk-5.1.1.tpg/m4/mpfr.m4
--- gawk-5.1.1/m4/mpfr.m4	2021-10-27 04:40:48.000000000 +0000
+++ gawk-5.1.1.tpg/m4/mpfr.m4	2021-11-04 09:00:36.241210163 +0000
@@ -18,7 +18,7 @@
      [_do_mpfr=$withval],[_do_mpfr=yes])
 
   if test "$_do_mpfr" != "no" ; then
-     if test -d "$withval" ; then
+     if test -d "$withval" && "$withval" != "/usr" ; then
         CPPFLAGS="${CPPFLAGS} -I$withval/include"
         LDFLAGS="${LDFLAGS} -L$withval/lib"
      fi
diff -Naur gawk-5.1.1/m4/readline.m4 gawk-5.1.1.tpg/m4/readline.m4
--- gawk-5.1.1/m4/readline.m4	2021-10-27 04:41:44.000000000 +0000
+++ gawk-5.1.1.tpg/m4/readline.m4	2021-11-04 08:59:55.761357471 +0000
@@ -24,7 +24,7 @@
      [_do_readline=$withval],[_do_readline=yes])
 
   if test "$_do_readline" != "no" ; then
-     if test -d "$withval" ; then
+     if test -d "$withval" && "$withval" != "/usr" ; then
         CPPFLAGS="${CPPFLAGS} -I$withval/include"
         LDFLAGS="${LDFLAGS} -L$withval/lib"
      fi
